src/app/profile/profile.component.js

```diff
@@ -169,51 +170,33 @@ class ProfileController {
         });
       }
       if(item.self && item.delete == undefined) { // update existing phone number
-        this.profileService.updatePhoneNumber(item)
-          .subscribe(
-            () => {
-              this.resetPhoneNumberForms();
-              this.success = true;
-              this.phonesLoading = false;
-            },
-            error => {
-              this.phoneNumberError = 'Failed updating phone numbers.';
-              this.$log.error(this.phoneNumberError, error.data);
-              this.phonesLoading = false;
-            }
-          );
+        requests.push(this.profileService.updatePhoneNumber(item));
       } else if(item.self && item.delete) { // delete existing phone number
-        this.profileService.deletePhoneNumber(item)
-          .subscribe(
-            () => {
-              pull(this.phoneNumbers, item);
-              this.resetPhoneNumberForms();
-              this.success = true;
-              this.phonesLoading = false;
-            },
-            error => {
-              this.phoneNumberError = 'Failed deleting phone numbers.';
-              this.$log.error(this.phoneNumberError, error.data);
-              this.phonesLoading = false;
-            }
-          );
-      } else if(!item.self && !item.delete){ // add new phone number
-        this.profileService.addPhoneNumber(item)
-          .subscribe(
-            data => {
-              data.spouse = item.spouse;
-              this.resetPhoneNumberForms();
-              this.success = true;
-              item['phone-number'] = data['phone-number'];
-              this.phonesLoading = false;
-            },
-            error => {
-              this.phoneNumberError = 'Failed adding phone numbers.';
-              this.$log.error(this.phoneNumberError, error.data);
-              this.phonesLoading = false;
-            }
-          );
+        requests.push(this.profileService.deletePhoneNumber(item)
+          .do(() => {
+            pull(this.phoneNumbers, item);
+          })
+        );
+      } else if(!item.self && !item.delete) { // add new phone number
+        requests.push(this.profileService.addPhoneNumber(item)
+          .do((data) => {
+            data.spouse = item.spouse;
```

Comment:
Does this do anything? Seems like nothing is using `data.spouse` here. If you did want to use a modified data object later in a subscribe or something I think you'd need to use `map` here instead of `do` and return the modified object. Maybe it would work with JS references but it wouldn't be very easy to figure out what's going on in the code.